client/my-sites/email/email-providers-stacked-comparison/provider-cards/google-workspace-card.tsx

```diff
@@ -74,23 +75,36 @@ const googleWorkspaceCardInformation: ProviderCard = {
 };
 
 const GoogleWorkspaceCard = ( {
+	cartDomainName,
 	comparisonContext,
-	currencyCode = '',
 	detailsExpanded,
-	domain,
-	gSuiteProductMonthly,
-	gSuiteProductYearly,
-	hasCartDomain,
 	intervalLength,
 	onExpandedChange,
 	selectedDomainName,
-	selectedSite,
-	shoppingCartManager,
 	source,
 }: EmailProvidersStackedCardProps ): ReactElement => {
 	const googleWorkspace: ProviderCard = { ...googleWorkspaceCardInformation };
 	googleWorkspace.detailsExpanded = detailsExpanded;
 
+	const hasCartDomain = Boolean( cartDomainName );
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+	const selectedSite = useSelector( getSelectedSite );
+	const domains = useSelector( ( state ) => getDomainsBySiteId( state, selectedSite?.ID ) );
+	const domain = getSelectedDomain( {
+		domains,
+		selectedDomainName: selectedDomainName,
+	} ) as ResponseDomain;
```

Comment:
Do we need this to be `as ResponseDomain`? I just took a closer look, and `getDomainsBySiteId()` currently returns `SiteDomain[]`, so it may be simpler to either update that type in a separate PR, or to just use `SiteDomain` as the type. (And yes, we need to coalesce those two types into a single type...)

Regardless, the "as X" coercion pattern is one I think we should generally try to avoid in cases where we have existing type information.